handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslClientContext.java

```diff
@@ -148,13 +151,13 @@ static OpenSslSessionContext newSessionContext(ReferenceCountedOpenSslContext th
         } catch (Exception e) {
             throw new SSLException("unable to setup trustmanager", e);
```

Comment:
now that `OpenSslKeyMaterialProvider` has an associated life cycle should we be more careful about cleaning the object up in the event of an exception? Perhaps we can clean it up in a finally block and null it out if we make it all the way through the method successfully and transfer the ownership to the return value.

(same comment on the server side).

Reply:
done...